### PR TITLE
Adding pointer and some focus events

### PR DESCRIPTION
**Pull request by MendyBerger**
This pr adds the following:

- Pointer events
- `FocusIn`/`Out` evnets
- `FocusEvent.related_target`

### CODE REVIEW

File: src/events.rs [hunk 1]
```diff
@@ -122,6 +122,31 @@ macro_rules! make_mouse_event {
...
 }
 
+macro_rules! make_pointer_event {
+    ($name:ident, $type:literal) => {
+        make_mouse_event!($name, $type => web_sys::PointerEvent);
+
+        impl $name {
+            #[inline] pub fn pointer_id(&self) -> i32 { self.event.pointer_id() }
+
+            #[inline] pub fn width(&self) -> i32 { self.event.width() }
```
Comment (Pauan) [inline src/events.rs:132]: Although this is called `width` in the DOM, I wonder if that's the best name.

Maybe it should be renamed to `pointer_width`, to make it more clear what it's doing?

File: src/events.rs [hunk 2]
```diff
@@ -122,6 +122,31 @@ macro_rules! make_mouse_event {
...
+            #[inline] pub fn height(&self) -> i32 { self.event.height() }
```
Comment (Pauan) [inline src/events.rs:133]: Same thing as for `width`

File: src/events.rs [hunk 3]
```diff
@@ -122,6 +122,31 @@ macro_rules! make_mouse_event {
...
+
+            #[inline] pub fn pressure(&self) -> f32 { self.event.pressure() }
+            #[inline] pub fn tangential_pressure(&self) -> f32 { self.event.tangential_pressure() }
+
+            #[inline] pub fn tilt_x(&self) -> i32 { self.event.tilt_x() }
+            #[inline] pub fn tilt_y(&self) -> i32 { self.event.tilt_y() }
+
+            #[inline] pub fn twist(&self) -> i32 { self.event.twist() }
+
+            #[inline] pub fn pointer_type(&self) -> String { self.event.pointer_type() }
```
Comment (Pauan) [inline src/events.rs:143]: This should probably return an enum. Since web-sys doesn't return an enum, maybe we should leave this out for now.

**Review by Pauan — CHANGES_REQUESTED** [review]

**Comment by MendyBerger** [comment]
@Pauan updated pr

**Review by Pauan — APPROVED** [review]
Thanks!

**Pull request merged**